### PR TITLE
feat: Implement `bool_then_to_if` assist

### PR DESCRIPTION
**Pull request by Veykril**
Other half of https://github.com/rust-analyzer/rust-analyzer/issues/8413

### CODE REVIEW

File: crates/ide_assists/src/handlers/convert_bool_then.rs [hunk 1]
```diff
@@ -102,6 +105,95 @@ pub(crate) fn convert_if_to_bool_then(acc: &mut Assists, ctx: &AssistContext) ->
...
+    let name_ref = ctx.find_node_at_offset::<ast::NameRef>()?;
+    let mcall = name_ref.syntax().parent().and_then(ast::MethodCallExpr::cast)?;
+    let receiver = mcall.receiver()?;
+    let closure_body = mcall.arg_list()?.args().exactly_one().ok()?;
+    let closure_body = match closure_body {
+        ast::Expr::ClosureExpr(expr) => expr.body()?,
+        _ => return None,
+    };
+    // Verify this is `bool::then` that is being called.
+    let func = dbg!(ctx.sema.resolve_method_call(&mcall))?;
```
Comment (lnicola) [inline crates/ide_assists/src/handlers/convert_bool_then.rs:138]:
```suggestion
    let func = ctx.sema.resolve_method_call(&mcall)?;
```

Comment (Veykril) [reply crates/ide_assists/src/handlers/convert_bool_then.rs:138]: I always miss a single one of those...

**Comment by Veykril** [comment]
bors retry

**Comment by lnicola** [comment]
bool-then-to-if

**Pull request merged**